public/js/src/module/store.js

```diff
@@ -667,32 +680,36 @@ function _firstItemOnly( results ) {
     }
 }
 
-function _transformExternalDataToXmlStr( survey ) {
-    if ( survey && survey.externalData ) {
-        survey.externalData = survey.externalData.map( instance => {
-            if ( instance.xml instanceof XMLDocument ) {
-                instance.xml = new XMLSerializer().serializeToString( instance.xml.documentElement, 'text/xml' );
-            }
+function _serializeSurvey( survey ) {
+    if ( survey ) {
+        if ( survey.externalData ) {
+            survey.externalData = survey.externalData.map( instance => {
+                if ( instance.xml instanceof XMLDocument ) {
+                    instance.xml = new XMLSerializer().serializeToString( instance.xml.documentElement, 'text/xml' );
+                }
 
-            return instance;
-        } );
-    }
+                return instance;
+            } );
+        }
 
-    return survey;
+        return encryptor.serializeEncryptedSurvey( survey );
+    }
 }
 
-function _transformExternalDataToXmlDoc( survey ) {
-    if ( survey && survey.externalData ) {
-        survey.externalData = survey.externalData.map( instance => {
-            if ( typeof instance.xml === 'string' ) {
-                instance.xml = parser.parseFromString( instance.xml, 'text/xml' );
-            }
+function _deserializeSurvey( survey ) {
+    if ( survey ) {
+        if ( survey.externalData ) {
+            survey.externalData = survey.externalData.map( instance => {
+                if ( typeof instance.xml === 'string' ) {
+                    instance.xml = parser.parseFromString( instance.xml, 'text/xml' );
+                }
 
-            return instance;
-        } );
-    }
+                return instance;
+            } );
+        }
 
-    return survey;
+        return encryptor.deserializeEncryptedSurvey( survey );
+    }
```

Comment:
In hindsight, I wish I had defined an actual `Survey` class which handles this (as well as documenting its actual type). I had also thought it might be possible to use a method similar to `toJSON`, but IndexedDB depends on serialization rules without an equivalent method. For future consideration, it might be good for our wrapper to provide a generalized (optional) mechanism for de/serialization of the types it handles.